gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1360 @@
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(String.format(
+          "Error when serializing change %s in managed zone %s in project %s.",
+          changeId, zoneName, projectId));
+    }
+  }
+
+  /**
+   * Returns a response to getZone service call.
+   */
+  Response getZone(String projectId, String zoneName, String[] fields) {
+    ZoneContainer container = findZone(projectId, zoneName);
+    if (container == null) {
+      return Error.NOT_FOUND.response(String.format(
+          "The 'parameters.managedZone' resource named '%s' does not exist.", zoneName));
+    }
+    ManagedZone result = OptionParsers.extractFields(container.zone(), fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(String.format(
+          "Error when serializing managed zone %s in project %s.", zoneName, projectId));
+    }
+  }
+
+  /**
+   * We assume that every project exists. If we do not have it in the collection yet, we just create
+   * a new default project instance with default quota.
+   */
+  Response getProject(String projectId, String[] fields) {
+    ProjectContainer defaultProject = createProject(projectId);
+    projects.putIfAbsent(projectId, defaultProject);
+    Project project = projects.get(projectId).project(); // project is now guaranteed to exist
+    Project result = OptionParsers.extractFields(project, fields);
+    try {
+      return new Response(HTTP_OK, jsonFactory.toString(result));
+    } catch (IOException e) {
+      return Error.INTERNAL_ERROR.response(
+          String.format("Error when serializing project %s.", projectId));
+    }
+  }
+
+  /**
+   * Creates a project. It generates a project number randomly (we do not have project numbers
+   * available).
+   */
+  private ProjectContainer createProject(String projectId) {
+    Quota quota = new Quota();
+    quota.setManagedZones(10000);
+    quota.setRrsetsPerManagedZone(10000);
+    quota.setRrsetAdditionsPerChange(100);
+    quota.setRrsetDeletionsPerChange(100);
+    quota.setTotalRrdataSizePerChange(10000);
+    quota.setResourceRecordsPerRrset(100);
+    Project project = new Project();
+    project.setId(projectId);
+    project.setNumber(new BigInteger(String.valueOf(
+        Math.abs(ID_GENERATOR.nextLong() % Long.MAX_VALUE))));
+    project.setQuota(quota);
+    return new ProjectContainer(project);
+  }
+
+  Response deleteZone(String projectId, String zoneName) {
+    ProjectContainer projectContainer = projects.get(projectId);
+    // check if empty
+    ZoneContainer zone = findZone(projectId, zoneName);
+    ImmutableList<RrsetWrapper> wrappers = zone == null
+        ? ImmutableList.<RrsetWrapper>of() : zone.dnsRecords().get();
+    for (RrsetWrapper current : wrappers) {
+      if (!ImmutableList.of("NS", "SOA").contains(current.rrset().getType())) {
```

Comment:
Move the creation of the list to outside the loop.  Or you could also make it a static class variable (if it's used anywhere else).